rpc/src/v1/impls/signing.rs

```diff
@@ -180,11 +174,16 @@ impl<D: Dispatcher + 'static> ParitySigning for SigningQueueClient<D> {
 
 	fn check_request(&self, id: RpcU256) -> Result<Option<RpcConfirmationResponse>, Error> {
 		let id: U256 = id.into();
-		match self.pending.lock().get(&id) {
-			Some(ref promise) => match promise.result() {
-				ConfirmationResult::Waiting => Ok(None),
-				ConfirmationResult::Rejected => Err(errors::request_rejected()),
-				ConfirmationResult::Confirmed(rpc_response) => rpc_response.map(Some),
+		match self.pending.lock().get_mut(&id) {
+			Some(ref mut future) => match future.poll() {
```

Comment:
fairly certain polling will panic if not done within the context of a task

Reply:
i suppose _technically_ it is part of a task because this function is future-ized by the RPC wrapper. but that's a sketchy invariant that isn't necessarily guaranteed to hold in the future.

Reply:
@tomusdrw wouldn't proper future-ization of signing RPCs not require `check_request` or other polling at all?

Reply:
@rphmeier as a second thought, I can cache the result of the polling as an `Option<ConfirmationResult>` and provide a method to look there — it will replace this polling here completely, and move the complexity from `check_request` into `ConfirmationReceiver`.

Seems to be a good idea, let my try that instead.